### PR TITLE
Altered retry timing parameters for connect to compute, to get more and quicker retries

### PR DESCRIPTION
**Pull request by jawj**
## Problem

Compute start time has improved, but the timing of connection retries from the proxy is rather slow, meaning we could be making clients wait hundreds of milliseconds longer than necessary.

## Summary of changes

Previously, retry time in ms was `100 * 1.5**n`, and `n` starts at 1, giving 150, 225, 337, 506, 759, 1139, 1709, ...

10 retries are specified, giving a total retry period of 17s.

This PR changes that to `25 * sqrt(2)**(n - 1)` instead, giving 25, 35, 50, 71, 100, 141, 200, ...

16 retries are specified, giving a total retry period of 15s.

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by jawj** [comment]
BTW, reviewers, please take a good look at this! I'm not really a Rust coder and don't have the full back-end set up on my machine, so all I know is that this compiles.

**Review by vadim2404 — APPROVED** [review]

**Comment by jawj** [comment]
I'm not clear why the v16 debug regress-tests are failing.

**Comment by bayandin** [comment]
> I'm not clear why the v16 debug regress-tests are failing.

Looks like a flaky test, restarted it.

**Review by conradludgate — APPROVED** [review]

File: proxy/src/proxy.rs [hunk 1]
```diff
@@ -554,7 +555,7 @@ impl ShouldRetry for compute::ConnectionError {
 
 pub fn retry_after(num_retries: u32) -> time::Duration {
     // 1.5 seems to be an ok growth factor heuristic
-    BASE_RETRY_WAIT_DURATION.mul_f64(1.5_f64.powi(num_retries as i32))
+    BASE_RETRY_WAIT_DURATION.mul_f64(RETRY_WAIT_EXPONENT_BASE.powi((num_retries - 1) as i32))
```
[the remark this answers is not in the document]
Comment (jawj) [reply proxy/src/proxy.rs:558]: Hmm. If we moved the cast as below then I think we solve that?

`BASE_RETRY_WAIT_DURATION.mul_f64(RETRY_WAIT_EXPONENT_BASE.powi((num_retries as i32) - 1))`

[the remark this answers is not in the document]
Comment (petuhovskiy) [reply proxy/src/proxy.rs:558]: Yeah, negative powers should do no harm, just can be a bit confusing.

> `BASE_RETRY_WAIT_DURATION.mul_f64(RETRY_WAIT_EXPONENT_BASE.powi((num_retries as i32) - 1))`

LGTM

**Review by petuhovskiy — APPROVED** [review]

**Pull request merged**